sycl/unittests/Extensions/CommandGraph/CommonReferenceSemantics.cpp

```diff
@@ -0,0 +1,194 @@
+//==----------------------- CommandGraph.cpp -------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#include "Common.hpp"
+
+#include <unordered_map>
+
+using namespace sycl;
+using namespace sycl::ext::oneapi;
+
+/**
+ * Checks that the operators and constructors of graph related classes meet the
+ * common reference semantics.
+ * @param objFactory A function object that returns an object to be tested.
+ */
+template <typename T, typename LambdaType>
+void testSemantics(LambdaType &&objFactory) {
```

Comment:
```suggestion
void testSemantics(LambdaType &&ObjFactory) {
```

and the doxy comment